google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineRunConfigurationEditor.java

```diff
@@ -84,25 +113,80 @@ private void onArtifactChanged() {
 
   protected void resetEditorFrom(CommonModel commonModel) {
     final AppEngineServerModel serverModel = (AppEngineServerModel) commonModel.getServerModel();
-    myPortField.setText(String.valueOf(serverModel.getLocalPort()));
     final Artifact artifact = serverModel.getArtifact();
     myArtifactComboBox.setSelectedItem(artifact);
     if (artifact == null && myArtifactComboBox.getItemCount() == 1) {
       myArtifactComboBox.setSelectedIndex(0);
+      BuildArtifactsBeforeRunTaskProvider.setBuildArtifactBeforeRun(
+          commonModel.getProject(), commonModel, (Artifact) myArtifactComboBox.getSelectedItem());
     }
-    myServerParametersEditor.setDialogCaption("Server Parameters");
-    myServerParametersEditor.setText(serverModel.getServerParameters());
+    port.setText(serverModel.getPort() != null
+        ? String.valueOf(serverModel.getPort()) : "");
+    enableAdvanced.setSelected(serverModel.getAdvancedSettings());
+    host.setText(serverModel.getHost());
+    adminHost.setText(serverModel.getAdminHost());
+    adminPort.setText(serverModel.getAdminPort() != null
+        ? String.valueOf(serverModel.getAdminPort()) : "");
+    authDomain.setText(serverModel.getAuthDomain());
+    storagePath.setText(serverModel.getStoragePath());
+    logLevel.setSelectedItem(serverModel.getLogLevel());
+    maxModuleInstances.setText(serverModel.getMaxModuleInstances() != null
+        ? String.valueOf(serverModel.getMaxModuleInstances()) : "");
+    useMtimeFileWatcher.setSelected(serverModel.getUseMtimeFileWatcher());
+    threadsafeOverride.setText(serverModel.getThreadsafeOverride());
+    jvmFlags.setDialogCaption(GctBundle.getString("appengine.run.jvmflags.title"));
+    jvmFlags.setText(Joiner.on(" ").join(serverModel.getJvmFlags()));
+    allowSkippedFiles.setSelected(serverModel.getAllowSkippedFiles());
+    apiPort.setText(serverModel.getApiPort() != null
+        ? String.valueOf(serverModel.getApiPort()) : "");
+    automaticRestart.setSelected(serverModel.getAutomaticRestart());
+    devappserverLogLevel.setSelectedItem(serverModel.getDevAppserverLogLevel());
+    skipSdkUpdateCheck.setSelected(serverModel.getSkipSdkUpdateCheck());
+    advancedSettingsPanel.setVisible(enableAdvanced.isSelected());
+    gcsBucketName.setText(serverModel.getDefaultGcsBucketName());
   }
 
   protected void applyEditorTo(CommonModel commonModel) throws ConfigurationException {
     final AppEngineServerModel serverModel = (AppEngineServerModel) commonModel.getServerModel();
+    serverModel.setPort(validateInteger(port.getText(), "port"));
+    serverModel.setArtifact(getSelectedArtifact());
+    serverModel.setAdvancedSettings(enableAdvanced.isSelected());
+
+    if (enableAdvanced.isSelected()) {
```

Comment:
What if the user enables advanced settings, modifies the config, and then hides the advanced settings. Would we not want those changes saved in this case?